### PR TITLE
Aqua/refactor - optimization

### PR DESCRIPTION
**Pull request by mingkang111**
# Description
This PR contains the first step of the refactoring work. Breaking down the change into step by step for convenient review. There will be another PR later for more refactor. To avoid breaking change, I haven't remove the original version for all moving. We will remove it in the last step.

# Major Change
* Provided centralized management for aqua job/model resources.
* Moved constants to `constants.py`.
* Moved all enums to `enums.py`. 
* Refactor `evaluation.py` to leverage new module.
* Restructure `evaluation.py`
```
aqua
|
--evaluation
       |
       --- __init__.py
       |
       --- evaluation.py
       |
       --- errors.py
       |
       --- entities.py
       |
       --- constants.py
```

# Next step
* Use wrapper in FT modules. 

# Test
Screenshot 2024-05-09 at 20 09 50
Screenshot 2024-05-09 at 20 10 54

### CODE REVIEW

**Comment by qiuosier** [comment]
Many tests are failing https://github.com/oracle/accelerated-data-science/actions/runs/9036162812/job/24832484555
Something wrong with mocking the `auth`?

**Comment by mingkang111** [comment]
> Many tests are failing https://github.com/oracle/accelerated-data-science/actions/runs/9036162812/job/24832484555 Something wrong with mocking the `auth`?

The failing test for opctl probably from merging conflict. I will update it one more time to check. 

The rest tests are failing in feature/aquav1.0.2 before this PR : https://github.com/oracle/accelerated-data-science/actions/runs/9025038228

File: ads/aqua/utils.py [hunk 1]
```diff
@@ -48,6 +47,22 @@
 EVALUATION_REPORT_MD = "report.md"
 EVALUATION_REPORT = "report.html"
 UNKNOWN_JSON_STR = "{}"
+FINE_TUNING_RUNTIME_CONTAINER = "iad.ocir.io/ociodscdev/aqua_ft_cuda121:0.3.17.20"
+DEFAULT_FT_BLOCK_STORAGE_SIZE = 750
+DEFAULT_FT_REPLICA = 1
+DEFAULT_FT_BATCH_SIZE = 1
+DEFAULT_FT_VALIDATION_SET_SIZE = 0.1
+HF_MODELS = "/home/datascience/conda/pytorch21_p39_gpu_v1/"
```
Comment (mayoor) [inline ads/aqua/utils.py:55]: Was this a temp constant? Do we still need to keep this?

File: ads/opctl/operator/lowcode/forecast/model/ml_forecast.py
Comment (mayoor) [inline ads/opctl/operator/lowcode/forecast/model/ml_forecast.py:1]: Let us not touch non-aqua files in this PR

Comment (mingkang111) [reply ads/opctl/operator/lowcode/forecast/model/ml_forecast.py:1]: Got it! It comes from resolving merge conflict. I will revert the change on the two files.

File: ads/aqua/utils.py (hunk 1 above)
Comment (mingkang111) [reply ads/aqua/utils.py:55]: This constant is not being used now. I will remove it.

**Pull request closed without merging**